### PR TITLE
Fix #5235: Tab Tray Menu Button Long Press Action Improvements

### LINKED ISSUE #5235: Tab Tray Menu Button Long Press Action Improvement
## Problem Description 
Duplicate tab and easier way to reach bookmark tab is asked by users. In order to achieve this easier way, adding "Duplicate Active Tab" and "Add Bookmark" actions should be added to tab menu button long press actions

## Design
Bookmark All Tab action title should be changed as "Add Bookmark for %i Tabs" indicating the total number of bookmarks and 
"Add Bookmark" action will be added in the same section.

Underneath Bookmark section, the "Duplicate Active Tab" should be added before Close Tab Section.

export

## Test Plan 

  - Open a new tab and check long press on tab tray icon in bottom toolbar
  - Press Duplicate Active Tab and check the actual tab url is opened in a new tab next to the active one
  
  - Long Press and check If Add Bookmark and Add Bookmark for %i Tabs buttons work as excpected
  
  "Add Bookmark" + "Duplicate Active Tab" should not exist in an empty open Tab.

### PR DESCRIPTION
**Pull request by soner-yuksel**
Adding "Add Bookmark" "Duplicate Active Tab" actions in long press Tab Tray icon in bottom toolbar.

## Summary of Changes

This pull request fixes #5235

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:

- Open a new tab and check long press on tab tray icon in bottom toolbar
- Press Duplicate Active Tab and check the actual tab url is opened in a new tab next to the active one
- Long Press and check If Add Bookmark and Add Bookmark for # Tabs buttons work as expected
- "Add Bookmark" + "Duplicate Active Tab" should not exist in an empty open Tab.

## Screenshots:

https://user-images.githubusercontent.com/6643505/162527303-bde0aec7-b727-497b-acf8-35e2f8525701.MP4

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: BraveShared/BraveStrings.swift [hunk 1]
```diff
@@ -128,7 +128,14 @@ extension Strings {
       value: "Are you sure you want to close all open tabs?",
       comment: "We ask users this prompt before attempting to close multiple tabs via context menu")
   public static let savedTabsFolderTitle = NSLocalizedString("SavedTabsFolderTitle", tableName: "BraveShared", bundle: Bundle.braveShared, value: "Saved Tabs", comment: "The title for the folder created when all bookmarks are being ")
-  public static let bookmarkAllTabsTitle = NSLocalizedString("BookmarkAllTabsTitle", tableName: "BraveShared", bundle: Bundle.braveShared, value: "Bookmark All Tabs", comment: "Action item title of long press for Adding Bookmark for All Tabs in Tab List")
+  public static let bookmarkAllTabsTitle = NSLocalizedString("BookmarkAllTabsTitle", tableName: "BraveShared", bundle: Bundle.braveShared, value: "Add Bookmark for %i Tabs", comment: "Action item title of long press for Adding Bookmark for All Tabs in Tab List - The parameter indicates the number of tabs like Add Bookmark for 5 Tabs")
```
Comment (kylehickinson) [inline BraveShared/BraveStrings.swift]: You may want `%lli` or `%lld` (they're the same for this case, but `%lld` is more common in this codebase) to ensure the formatting works correctly with large numbers possible from Swift's `Int`

File: BraveShared/BraveStrings.swift [hunk 2]
```diff
@@ -128,7 +128,14 @@ extension Strings {
...
+  public static let duplicateActiveTab =
+    NSLocalizedString(
+      "DuplicateActiveTab",
+      tableName: "BraveShared",
+      bundle: Bundle.braveShared,
+      value: "Duplicate Active Tabs",
```
Comment (kylehickinson) [inline BraveShared/BraveStrings.swift]: This is plural ("Tabs") but in the video it shows "Tab", is it supposed to be Tab here?

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 3]
```diff
@@ -2471,10 +2469,23 @@ extension BrowserViewController: TabManagerDelegate {
...
+        })
 
-    if tabManager.openedWebsitesCount > 0 {
+      bookmarkMenuChildren.append(bookmarkActiveTab)
+    }
+    
+    if tabManager.tabsForCurrentMode.count > 1 {
       let bookmarkAllTabs = UIAction(
-        title: Strings.bookmarkAllTabsTitle,
+        title: String(format: Strings.bookmarkAllTabsTitle, tabManager.tabsForCurrentMode.count),
```
Comment (kylehickinson) [inline Client/Frontend/Browser/BrowserViewController.swift]: Doesn't matter for this string in particular, but you should use `String.localizedStringWithFormat(...)` instead for localized strings as they take into account formatting order if the translator adds them

File: Client/Frontend/Browser/Tab.swift [hunk 4]
```diff
@@ -178,6 +178,14 @@ class Tab: NSObject {
   var isDesktopSite: Bool {
     webView?.customUserAgent?.lowercased().contains("mobile") == false
   }
+  
+  var containsWebPage: Bool {
+    if let url = url {
+      return url.isWebPage() && !(InternalURL(url)?.isAboutHomeURL ?? false)
```
Comment (kylehickinson) [inline Client/Frontend/Browser/Tab.swift]: `!(InternalURL(url)?.isAboutHomeURL ?? false)` this is very confusing to read haha, maybe we can do something like

```swift
let isHomeURL = InternalURL(url)?.isAboutHomeURL
return url.isWebPage && isHomeURL != true
```

**Review by kylehickinson — CHANGES_REQUESTED** [review]

File: BraveShared/BraveStrings.swift (hunk 1 above)
Comment (soner-yuksel) [reply BraveShared/BraveStrings.swift]: Sounds good Fixed at https://github.com/brave/brave-ios/pull/5236/commits/013a9ecefd1849141e901f1d9fb053db9ca938cf

File: BraveShared/BraveStrings.swift (hunk 2 above)
Comment (soner-yuksel) [reply BraveShared/BraveStrings.swift]: Fix at https://github.com/brave/brave-ios/pull/5236/commits/013a9ecefd1849141e901f1d9fb053db9ca938cf

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 3 above)
Comment (soner-yuksel) [reply Client/Frontend/Browser/BrowserViewController.swift]: Fixed at https://github.com/brave/brave-ios/pull/5236/commits/013a9ecefd1849141e901f1d9fb053db9ca938cf

File: Client/Frontend/Browser/Tab.swift (hunk 4 above)
Comment (soner-yuksel) [reply Client/Frontend/Browser/Tab.swift]: But you like the optional dance in Swift :P Fixed at https://github.com/brave/brave-ios/pull/5236/commits/013a9ecefd1849141e901f1d9fb053db9ca938cf

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 5]
```diff
@@ -2485,6 +2496,23 @@ extension BrowserViewController: TabManagerDelegate {
...
       bookmarkMenuChildren.append(bookmarkAllTabs)
     }
+    
+    var duplicateTabMenuChildren: [UIAction] = []
+
+    if containsWebPage, let selectedTab = tabManager.selectedTab, let url = selectedTab.fetchedURL {
+      let duplicateActiveTab = UIAction(
+        title: Strings.duplicateActiveTab,
+        image: UIImage(systemName: "plus.square.on.square"),
+        handler: UIAction.deferredActionHandler { _ in
```
Comment (kylehickinson) [inline Client/Frontend/Browser/BrowserViewController.swift]: does this need `[unowned self]` as well?

**Review by kylehickinson — APPROVED** [review]
Just one minor thing

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 5 above)
Comment (soner-yuksel) [reply Client/Frontend/Browser/BrowserViewController.swift]: Fixed at https://github.com/brave/brave-ios/pull/5236/commits/0d2fe3d86d839c390e029f0d78408ae0967f5d5d

**Pull request merged**